Changes in module 'nf-core/meryl/unionsum'
--- modules/nf-core/meryl/unionsum/meta.yml
+++ modules/nf-core/meryl/unionsum/meta.yml
@@ -36,8 +36,9 @@
   - meryl_db:
       type: directory
       description: A Meryl k-mer database that is the union sum of the input databases
-      pattern: "*.unionsum.meryldb"
+      pattern: "*.unionsum.meryl"
 authors:
   - "@mahesh-panchal"
 maintainers:
   - "@mahesh-panchal"
+  - "@gallvp"

--- modules/nf-core/meryl/unionsum/main.nf
+++ modules/nf-core/meryl/unionsum/main.nf
@@ -12,7 +12,7 @@
     val kvalue
 
     output:
-    tuple val(meta), path("*.unionsum.meryldb"), emit: meryl_db
+    tuple val(meta), path("*.unionsum.meryl"), emit: meryl_db
     path "versions.yml"                        , emit: versions
 
     when:
@@ -25,8 +25,9 @@
     meryl union-sum \\
         k=$kvalue \\
         threads=$task.cpus \\
+        memory=${task.memory.toGiga()} \\
         $args \\
-        output ${prefix}.unionsum.meryldb \\
+        output ${prefix}.unionsum.meryl \\
         $meryl_dbs
 
     cat <<-END_VERSIONS > versions.yml
@@ -39,7 +40,7 @@
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    touch ${prefix}.unionsum.meryldb
+    touch ${prefix}.unionsum.meryl
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":

--- modules/nf-core/meryl/unionsum/tests/main.nf.test.snap
+++ modules/nf-core/meryl/unionsum/tests/main.nf.test.snap
@@ -8,7 +8,7 @@
                             "id": "test",
                             "single_end": true
                         },
-                        "test.unionsum.meryldb:md5,d41d8cd98f00b204e9800998ecf8427e"
+                        "test.unionsum.meryl:md5,d41d8cd98f00b204e9800998ecf8427e"
                     ]
                 ],
                 "1": [
@@ -20,7 +20,7 @@
                             "id": "test",
                             "single_end": true
                         },
-                        "test.unionsum.meryldb:md5,d41d8cd98f00b204e9800998ecf8427e"
+                        "test.unionsum.meryl:md5,d41d8cd98f00b204e9800998ecf8427e"
                     ]
                 ],
                 "versions": [
@@ -32,7 +32,7 @@
             "nf-test": "0.8.4",
             "nextflow": "23.10.1"
         },
-        "timestamp": "2024-03-27T10:19:52.291287389"
+        "timestamp": "2024-05-22T12:40:21.306142"
     },
     "sarscov2 - fastq - single_end": {
         "content": [

************************************************************
