Changes in module 'nf-core/merqury'
--- modules/nf-core/merqury/main.nf
+++ modules/nf-core/merqury/main.nf
@@ -26,6 +26,7 @@
     tuple val(meta), path("${prefix}.qv")        , emit: assembly_qv
     tuple val(meta), path("${prefix}.*.qv")      , emit: scaffold_qv
     tuple val(meta), path("*.hist.ploidy")       , emit: read_ploidy
+    tuple val(meta), path("*.hapmers.blob.png")  , emit: hapmers_blob_png           , optional: true
     path "versions.yml"                          , emit: versions
 
     when:
@@ -56,4 +57,30 @@
         merqury: $VERSION
     END_VERSIONS
     """
+
+    stub:
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = 1.3
+    """
+    touch ${prefix}_only.bed
+    touch ${prefix}_only.wig
+    touch ${prefix}.completeness.stats
+    touch ${prefix}.dist_only.hist
+    touch ${prefix}.spectra-cn.fl.png
+    touch ${prefix}.spectra-cn.hist
+    touch ${prefix}.spectra-cn.ln.png
+    touch ${prefix}.spectra-cn.st.png
+    touch ${prefix}.spectra-asm.fl.png
+    touch ${prefix}.spectra-asm.hist
+    touch ${prefix}.spectra-asm.ln.png
+    touch ${prefix}.spectra-asm.st.png
+    touch ${prefix}.qv
+    touch ${prefix}.${prefix}.qv
+    touch ${prefix}.hist.ploidy
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        merqury: $VERSION
+    END_VERSIONS
+    """
 }

************************************************************
