Changes in module 'nf-core/meryl/count'
--- modules/nf-core/meryl/count/meta.yml
+++ modules/nf-core/meryl/count/meta.yml
@@ -38,8 +38,9 @@
   - meryl_db:
       type: directory
       description: A Meryl k-mer database
-      pattern: "*.meryldb"
+      pattern: "*.meryl"
 authors:
   - "@mahesh-panchal"
 maintainers:
   - "@mahesh-panchal"
+  - "@gallvp"

--- modules/nf-core/meryl/count/main.nf
+++ modules/nf-core/meryl/count/main.nf
@@ -12,8 +12,8 @@
     val kvalue
 
     output:
-    tuple val(meta), path("*.meryldb"), emit: meryl_db
-    path "versions.yml"               , emit: versions
+    tuple val(meta), path("*.meryl")    , emit: meryl_db
+    path "versions.yml"                 , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
@@ -26,9 +26,10 @@
         meryl count \\
             k=$kvalue \\
             threads=$task.cpus \\
+            memory=${task.memory.toGiga()} \\
             $args \\
             $reads \\
-            output read.\${READ%.f*}.meryldb
+            output read.\${READ%.f*}.meryl
     done
 
     cat <<-END_VERSIONS > versions.yml
@@ -42,7 +43,7 @@
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     for READ in $reads; do
-        touch read.\${READ%.f*}.meryldb
+        touch read.\${READ%.f*}.meryl
     done
 
     cat <<-END_VERSIONS > versions.yml

--- modules/nf-core/meryl/count/tests/main.nf.test.snap
+++ modules/nf-core/meryl/count/tests/main.nf.test.snap
@@ -303,7 +303,7 @@
                             "id": "test",
                             "single_end": true
                         },
-                        "read.test1_1.meryldb:md5,d41d8cd98f00b204e9800998ecf8427e"
+                        "read.test1_1.meryl:md5,d41d8cd98f00b204e9800998ecf8427e"
                     ]
                 ],
                 "1": [
@@ -315,7 +315,7 @@
                             "id": "test",
                             "single_end": true
                         },
-                        "read.test1_1.meryldb:md5,d41d8cd98f00b204e9800998ecf8427e"
+                        "read.test1_1.meryl:md5,d41d8cd98f00b204e9800998ecf8427e"
                     ]
                 ],
                 "versions": [
@@ -327,6 +327,6 @@
             "nf-test": "0.8.4",
             "nextflow": "23.10.1"
         },
-        "timestamp": "2024-03-26T19:30:13.745468405"
+        "timestamp": "2024-05-22T12:27:16.300916"
     }
 }
************************************************************
